### PR TITLE
Add test for convert function

### PR DESCRIPTION
**Pull request by vmpyr**
test added at `test/strings/basic.jl` for function usage at `base/strings/substring.jl` line 62 and 63

### CODE REVIEW

File: test/strings/basic.jl [hunk 1]
```diff
@@ -938,6 +938,17 @@ end
     end
 end
 
+@testset "Conversion to Type{Union{String, SubString{String}}}" begin
+    str::String = "abc"
```
Comment (KristofferC) [inline test/strings/basic.jl:942]: This `::String` seems redundant?

File: test/strings/basic.jl [hunk 2]
```diff
@@ -938,6 +938,17 @@ end
...
+    substr = SubString(str, 1, 3)
```
Comment (KristofferC) [inline test/strings/basic.jl:943]: This is just `SubString(str)`?

File: test/strings/basic.jl [hunk 3]
```diff
@@ -938,6 +938,17 @@ end
...
+    for T in [typeof(str), typeof(substr)]
```
Comment (KristofferC) [inline test/strings/basic.jl:944]: Might as well write this out explicitly:

```suggestion
    for T in [String, SubString{String}]
```

File: test/strings/basic.jl [hunk 4]
```diff
@@ -938,6 +938,17 @@ end
...
+        conv_str = convert(T, str)
+        conv_substr = convert(T, substr)
+        @test typeof(conv_str) == T
```
Comment (KristofferC) [inline test/strings/basic.jl:947]: This doesn't test that the result of the conversion is good, only the type.

File: test/strings/basic.jl (hunk 1 above)
Comment (vmpyr) [reply test/strings/basic.jl:942]: I wrote it just to be sure and also as a way of documenting

File: test/strings/basic.jl (hunk 2 above)
Comment (vmpyr) [reply test/strings/basic.jl:943]: Will do

File: test/strings/basic.jl (hunk 3 above)
Comment (vmpyr) [reply test/strings/basic.jl:944]: Okay.

**Pull request merged**